scripts/data_overlap/compute_data_overlap_metrics.py

```diff
@@ -130,69 +109,41 @@ def create_ngram_index(
                     for reference_ngram in ngrams(reference_unigrams, n):
                         if reference_ngram not in ngram_index[n]:
                             ngram_index[n][reference_ngram] = set()
+                        id = instance.id if instance.id else f"id{i}"
                         ngram_index[n][reference_ngram].add(
-                            EntryDataOverlapKey(stats_key=stats_key, instance_id=i, part=PART_REF)
+                            EntryDataOverlapKey(stats_key=stats_key, instance_id=id, index=i, part=PART_REF)
                         )
     return ngram_index
 
 
-def create_all_data_overlap_stats(light_scenarios: List[LightScenario], n_values: List[int]) -> AllDataOverlapStats:
-    """Given a list of scenarios and n values, initialize all_overlap_stats"""
-    hlog("Initializing all data overlap stats")
-    all_overlap_stats: AllDataOverlapStats = {}
-    for scenario in light_scenarios:
-        for n in n_values:
-            # Initialize a stats instance for every pair of <scenario, n>
-            stats: DataOverlapStats = DataOverlapStats.from_scenario(scenario, stats_tags={"N": n})
-            if stats.stats_key in all_overlap_stats:
-                raise ValueError("Duplicated settings detected.")
-            all_overlap_stats[stats.stats_key] = stats
-    return all_overlap_stats
-
-
-def compute_scenario_file_data_overlap(
+def compute_all_data_overlap(
     training_file_path: str,
     file_format: str,
     ngram_index: NgramIndex,
-    all_overlap_stats: AllDataOverlapStats,
     tokenizer: LightTokenizer,
-    ngram_counter: Optional[NgramCounter] = None,
-    max_overlapping_ngrams: int = 0,
+    stats_key_to_input_ids: Dict[DataOverlapStatsKey, Set[str]],
+    stats_key_to_reference_ids: Dict[DataOverlapStatsKey, Set[str]],
 ):
     """
-    Given an input file, compute a overlap stats for each n and each scenario by calling
-    `compute_scenario_document_overlap()` for each document in the file. The function writes
-    to the overlap stats directly and does not return anything.
-
-    ngram_index: The ngram index that maps from ngrams to overlap stats
-
-    all_overlap_stats: The overlap stats for each scenario and n. The variable to write to.
-
-    tokenizer: The tokenizer used to break documents in the file into tokens
-
-    ngram_counter: The ngrams that are overlapped between the training file and the scenario data
-    and their counts.
-    The outer dict maps from n to the inner dict, which maps from ngram to count.
+    Iterate through documents to compute overlaps
```

Comment:
nit: quotes should be on same line for one-line docstrings as per [PEP 257](https://peps.python.org/pep-0257/#one-line-docstrings) `"""Iterate through documents to compute overlaps."""`